diff --git a/dist/esm/wrap.js b/dist/esm/wrap.js
index 6ce8027deb0f211098ec0ce79bc38f44bd7f072b..a3f666ca3d5d8a7738044c865578a96bc741ca44 100644
--- a/dist/esm/wrap.js
+++ b/dist/esm/wrap.js
@@ -11,6 +11,7 @@ export function wrap(item, handler, autoPassthrough = true) {
     }
     const proxyHandler = Object.assign({}, handler);
     proxyHandler.get = (target, prop, receiver) => {
+        if (prop ==='BROWSER') return target.BROWSER
         if (prop === unwrapSymbol) {
             return item;
         }