diff --git a/data/createRouteAction.tsx b/data/createRouteAction.tsx
index 91c1ca2d061e64326ba50d14cc5d6066a0822acd..33254b048f83273b9d146dfab50f383a25a029d7 100644
--- a/data/createRouteAction.tsx
+++ b/data/createRouteAction.tsx
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import { useNavigate, useSearchParams, type Navigator } from "@solidjs/router";
 import { $TRACK, batch, createSignal, useContext } from "solid-js";
 import { FormError, FormImpl, FormProps } from "./Form";
diff --git a/entry-client/mount.tsx b/entry-client/mount.tsx
index d4194c13e60adbe9fdf22dd421696add5b264e97..7584228473a8d4e6264a32f56d8eb4230988c105 100644
--- a/entry-client/mount.tsx
+++ b/entry-client/mount.tsx
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import type { JSX } from "solid-js";
 import { getOwner } from "solid-js";
 import { createComponent, getNextElement, hydrate, render } from "solid-js/web";
diff --git a/islands/router.ts b/islands/router.ts
index 6ee641e49b7efc64d3ef0bf2992b7d4c9119f981..9a47639642563faaf2c9eac0b2136d76dd945e9f 100644
--- a/islands/router.ts
+++ b/islands/router.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import type { Location, Navigator } from "@solidjs/router";
 import { createSignal } from "solid-js";
 interface LocationEntry {
diff --git a/server/middleware.ts b/server/middleware.ts
index 610d050a9e9005ceb7f45a24c6596381e5b80718..5185ebdbedf60d5fd7fdf2d10080a5d66c6951c7 100644
--- a/server/middleware.ts
+++ b/server/middleware.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import { internalFetch } from "../api/internalFetch";
 import { Middleware as ServerMiddleware } from "../entry-server/StartServer";
 import { ContentTypeHeader, XSolidStartContentTypeHeader, XSolidStartOrigin } from "./responses";
diff --git a/server/server-functions/server.ts b/server/server-functions/server.ts
index d09165c339aa0db4c865ca83da65505c1e9234da..bb7646665a34e30b2fde90366040e69db7a7acff 100644
--- a/server/server-functions/server.ts
+++ b/server/server-functions/server.ts
@@ -1,3 +1,4 @@
+// @ts-nocheck
 import { sharedConfig } from "solid-js";
 import { internalFetch } from "../../api/internalFetch";
 import { FormError } from "../../data";